### ISSUE TITLE
Can the code be used to generate the triggers for the vision transformer?

### ISSUE THREAD

**Issue by mo666666** [YiZeng623/frequency-backdoor#3]
Hi, Yizeng. Thank you for your excellent work. Can your code be used to generate trigger for another architecture, e.g. ViT? Looking forward to your reply!

**Comment by YiZeng623** [comment]
Hi there. Thanks for your interest in this work. 

FYI, guess you are talking about Section 5 to generate the smooth trigger. The code can directly be used to generalize to ViT if your model structures are defined in TensorFlow. You can do it in PyTorch as well, which is also quite simple: {load your model (ViT), plug in a PGD attack workflow in PyTorch, replace the gradient projection in PGD with the low-pass filter}, then you should be good to go.

However, one thing I would like to clarify is that the generated trigger for CIFAR-10 and GTSRB should enjoy quite well generalizability to attack different structures. You might not need to generate a trigger dedicated to ViTs.

Some side notes: Our recent work developed an [in-ward pointing design of a clean-label backdoor attack](https://arxiv.org/pdf/2204.05255.pdf). Given the same low-pass filter as the constraint, the new attack is more effective than the smooth trigger developed in this paper with better stealthiness (lower poison ratio/ clean-label attack/ higher ASR).

**Comment by mo666666** [comment]
Yizeng, thank you for your reply!!! Your suggestion is quite helpful!!!
